radicale/__init__.py

```diff
@@ -425,6 +433,13 @@ def response(status, headers=(), answer=None):
                     environ, base_prefix, path, user)
             except socket.timeout:
                 return response(*REQUEST_TIMEOUT)
+            except vobject.base.VObjectError as e:
+                return response(*UNPROCESSABLE_ENTITY(e))
```

Comment:
I don't think that 422 is the right error. It's for semantic errors, 400 should be more appropriate.

Why didn't you put it into do_PUT, where uploaded data from the client is parsed? Here it catches all errors from vobject, no matter where they happen.